docs/guides/ajax.rst

```diff
@@ -1,31 +1,220 @@
+
+
+Fetching data
+-------------
+
+To fetch arbitrary data from registered page handlers, use ``fetchPath('url_path', options)``.
+
+.. code-block:: php
+
+	// in myplugin/start.php
+	elgg_register_page_handler('myplugin_time', 'myplugin_get_time');
+
+	function myplugin_get_time() {
+		elgg_ajax_gatekeeper();
+
+		echo json_encode([
+			'rfc2822' => date(DATE_RFC2822),
+			'day' => date('l'),
+		]);
+
+		return true;
+	}
+
+.. code-block:: js
+
+	var ajax = new require('elgg/Ajax')();
+
+	ajax.fetchPath('myplugin_time').done(function (output) {
+		alert(output.rfc2822);
+		alert(output.day);
+	});
+
+Notes for paths:
+
+* All hooks have type ``path:<url_path>``. In this case, "path:<myplugin_time>".
+* If the page handler echoes a regular web page, ``output`` will be a string containing the HTML.
+
+Fetching views
+--------------
+
+To fetch a server-rendered view, use ``fetchView('view_name', options)``. If the view is a PHP file, you must register it for Ajax first:
+
+.. code-block:: php
+
+	// in myplugin_init()
+	elgg_register_ajax_view('myplugin/get_link');
+
+.. code-block:: php
+
+	// in myplugin/views/default/myplugin/get_link.php
+
+	if (empty($vars['entity']) || !$vars['entity'] instanceof ElggObject) {
+		return;
+	}
 
-.. code:: js
+	$object = $vars['entity'];
+	/* @var ElggObject $object */
+
+	echo elgg_view('output/url', [
+		'text' => $object->getDisplayName(),
+		'href' => $object->getUrl(),
+		'is_trusted' => true,
+	]);
+
+.. code-block:: js
+
+	var ajax = new require('elgg/Ajax')();
+
+	ajax.fetchView('myplugin/get_link', {
+		data: {
+			guid: 123 // querystring
+		},
+	}).done(function (output) {
+		$('.myplugin-link').html(output);
+	});
+
+Notes for views:
+
+* All hooks have type ``view:<view_name>``. In this case, "view:myplugin/get_link".
+* ``output`` will be a string with the rendered view.
+* The request data are injected into ``$vars`` in the view.
+* If the request data contains ``guid``, the system sets ``$vars['entity']`` to the corresponding entity or ``false`` if it can't be loaded.
+
+.. warning::
+
+	Ajax views must treat ``$vars`` as completely untrusted user data. You may want to use ``get_input()``
+	instead to make this clear to other developers. Review the use of ``$vars`` in an existing view before
+	registering it for Ajax fetching.
+
+
+Fetching forms
+--------------
+
+Fetch server-rendered forms using ``fetchForm('action_name', options)``. You must register the view for Ajax first:
+
+.. code-block:: php
+
+	// in myplugin_init()
+	elgg_register_ajax_view('forms/myplugin/add');
+
+.. code-block:: js
+
+	var ajax = new require('elgg/Ajax')();
+
+	ajax.fetchForm('myplugin/add').done(function (output) {
+		$('.myplugin-form-container').html(output);
+	});
+
+Notes for forms:
+
+* All hooks have type ``form:<action_name>``. In this case, "form:myplugin/add".
+* ``output`` will be a string with the rendered view.
+* The request data are injected into ``$vars`` in your view.
```

Comment:
I never used it, and I don't I think I ever will. I use full URL path, e.g. `blog/edit/$guid`, to make sure all the page owner and context logic stays intact, and only output the form from the page handler if it's ajax request.